packages/next/src/build/swc/index.ts

```diff
@@ -824,7 +834,7 @@ function bindingToApi(
               getReactCompilerLoader(
                 reactCompilerOptions,
                 projectPath,
-                nextConfig.dev,
```

Comment:
This is the origin of this change. `nextConfig.dev` (or any unknown property) is typed as `any` so we didn't know `undefined` flowed into the  Babel plugin options.